trellis-http/src/main/java/org/trellisldp/http/impl/PutHandler.java

```diff
@@ -236,8 +237,11 @@ public ResponseBuilder setResource(final Resource res) {
                         .forEachOrdered(dataset::add);
                 }
             });
-
-            if (resourceService.put(internalId, ldpType, dataset.asDataset()).get()) {
+            // TODO is this the best we can do? what concurrency errors are lurking?
```

Comment:
it is possible for two nodes to have simultaneous `::create` operations. It is therefore possible for the persistence layer to duplicate containment triples on the parent, but it is also possible for that layer to deal intelligently with that condition. E.g. the kafka-based impl uses ZK as a coordinating agent to prevent against that condition. And for synchronous implementations, the tricky parts are not so hard to avoid (at least they weren't for the triplestore impl).

Reply:
Yeah, the question for me is how much the upper layers (above `ResourceService`) want to do about that. My inclination is: not very much at all!

Reply:
Yes, I completely agree. I don't want to see any locking/coordinating code anywhere in the `-api` or `-http` layer.